proto/proto-backplane-grpc/src/main/proto/deephaven/proto/storage.proto

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+syntax = "proto3";
+
+package io.deephaven.proto.backplane.grpc;
+
+option java_multiple_files = true;
+option optimize_for = SPEED;
+option go_package = "github.com/deephaven/deephaven-core/go/internal/proto/storage";
+
+/*
+ * Shared storage management service.
+*/
+service StorageService {
+  rpc ListItems(ListItemsRequest) returns (ListItemsResponse) {}
+  rpc FetchFile(FetchFileRequest) returns (FetchFileResponse) {}
+  // Can create new files or modify existing
+  rpc SaveFile(SaveFileRequest) returns (SaveFileResponse) {}
+  rpc MoveItem(MoveItemRequest) returns (MoveItemResponse) {}
+  rpc CreateDirectory(CreateDirectoryRequest) returns (CreateDirectoryResponse) {}
+  rpc DeleteItem(DeleteItemRequest) returns (DeleteItemResponse) {}
+}
+
+message ListItemsRequest {
+  string path = 1;// the path to the directory to list. empty to list top level
+  optional string filter_glob = 2;// A pattern to filter for, with "*" as a wildcard
+}
+
+enum FileKind {
+  UNKNOWN = 0;// Should not be used, exists only to indicate that this was unset
+  DIRECTORY = 1;
+  FILE = 2;
+}
+
+message FileInfo {
+  string path = 1;
```

Comment:
Can you elaborate? Are you suggesting a filesystem that we would interact with using non-unicode paths, i.e. either "ascii-only" or "there exist valid bytes that can be in a path that are not valid utf8"?

I don't think either case makes sense, at least from the perspective of "the server is written in Java, and Java's filesystem APIs all use Strings, which implies unicode". A quick read of sun.nio.fs.UnixPath shows that while paths are encoded as bytes, null bytes are disallowed, and a specific charset (provided by system property `sun.jnu.encoding`) will be used to do the actual encoding/decoding. https://stackoverflow.com/a/63688076/860630 suggests that this is "the default charset" on linux at least, and as of recent java versions, that means utf-8 (as of java 18).

Reply:
The "there exist valid bytes that can be in a path that are not valid utf8" part - technically, Unix-like OSes, paths are *bytes*. It's a fact that I think a lot of APIs overlook. https://jod.al/2019/12/10/pathlib-and-paths-with-arbitrary-bytes/

Reply:
Technically correct, but is it a useful distinction? Are there appropriate java apis to interact with the file system in this way?

Reply:
> but a useful distinction?

"maybe"?

I am being pedantic here. I'm not too concerned with the java APIs - if we move path to `bytes`, we'll assert UTF-8. I'm thinking about implementations beyond java and potential breaking changes we would need otherwise in the future. I know some languages expose byte-oriented path apis.

This was actually from somebody I know from college: https://beets.io/blog/paths.html